### PR TITLE
App replacement

### PR DESCRIPTION
**Pull request by AustinSeto**
https://github.com/OpenLiberty/guide-microprofile-rest-client-async/issues/53

### CODE REVIEW

**Comment by gkwan-ibm** [comment]
hi @Charlotte-Holt would you review the README.adoc?

File: README.adoc [hunk 1]
```diff
@@ -33,37 +33,45 @@ Learn how to use MicroProfile Rest Client to invoke RESTful microservices asynch
...
+The `system` and `inventory` microservices use MicroProfile Reactive Messaging to send and receive the system load events. 
+If you want to learn more about reactive messaging, you can read the https://openliberty.io/guides/microprofile-reactive-messaging.html[Creating Reactive Java Microservices^] guide.
 
-If you want to learn more about MicroProfile Rest Client, you can read the https://openliberty.io/guides/microprofile-rest-client.html[Consuming RESTful services with template interfaces^] guide.
+// =================================================================================================
+// Prerequisites
+// =================================================================================================
+== Additional prerequisites
 
+You need to have Docker installed. For installation instructions, refer to the official https://docs.docker.com/get-docker/[Docker documentation^]. You will build and run the microservices in Docker containers. An installation of Apache Kafka is provided in another Docker container.
```
Comment (Charlotte-Holt) [inline README.adoc:74]: Suggested update

>You need to have Docker installed.

to

>Before you begin, Docker needs to be installed.

File: README.adoc [hunk 2]
```diff
@@ -76,136 +84,173 @@ include::{common-includes}/gitclone.adoc[]
...
+The `CompletionStage` interface represents a unit of computation.
+After a computation is complete, it can be either finished or chained with more completion stages using the [hotspot=thenAcceptAsync file=0]`thenAcceptAsync()` method. 
+Exceptions are handled in a callback provided to the [hotspot=exceptionally file=0]`exceptionally()` method, which behaves similar to a catch block.
+When you return a `CompletionStage` type in the resource, it doesn’t necessarily mean that the computation completed and the response was built.
+JAX-RS will respond to the caller after the `CompletionStage` completes.
+
+A [hotspot=countdown1 hotspot=countdown2 hotspot=countdown3 file=0]`CountDownLatch` is used to keep track of how many asynchronous requests we are waiting on.
+Using the [hotspot=await file=0]`await` method, we stop and wait until all our requests are complete.
```
[the remark this answers is not in the document]
Comment (AustinSeto) [reply README.adoc:132]: Hi Charlotte, perhaps the following rewording works?

> By using the [hotspot=await file=0]await method, the program stops and waits until all requests are complete.

File: README.adoc [hunk 3]
```diff
@@ -76,136 +84,173 @@ include::{common-includes}/gitclone.adoc[]
...
+
+A [hotspot=countdown1 hotspot=countdown2 hotspot=countdown3 file=0]`CountDownLatch` is used to keep track of how many asynchronous requests we are waiting on.
+Using the [hotspot=await file=0]`await` method, we stop and wait until all our requests are complete. 
 
-The `CompletionStage` interface represents a unit of computation. After that computation completes, it can either be finished or chained with more completion stages by using the [hotspot=thenApplyAsync file=0]`thenApplyAsync()` method to perform more computations. Exceptions can be handled in a callback provided to the [hotspot=exceptionally file=0]`exceptionally()` method, which behaves similar to a catch block. When you return a `CompletionStage` type in the resource, it doesn’t necessarily mean that the computation completed and the response was built. JAX-RS will respond to the caller after the `CompletionStage` completes.
+Delcaring the `values` variable with [hotspot=volatile file=0]`volatile` and instantiating it as a [hotspot=concurrentHashMap file=0]`ConcurrentHashMap` allows it to store system information and safely access it asynchronously from multiple threads.
```
[the remark this answers is not in the document]
Comment (AustinSeto) [reply README.adoc:134]: Updated to the following:

> Declaring the `values` variable with the `volatile` keyword and instantiating it as a `ConcurrentHashMap` object allows it to store system information and safely access that information asynchronously from multiple threads.

[the remark this answers is not in the document]
Comment (AustinSeto) [reply README.adoc]: Updated to the following:

> Declaring the `values` variable with the `volatile` keyword and instantiating it as a `ConcurrentHashMap` object allows it to store system information and safely access that information asynchronously from multiple threads.

**Comment by AustinSeto** [comment]
Hi @Charlotte-Holt I've made and pushed all the changes you've suggested except for these two. 

For 

> > Using the [hotspot=await file=0]`await` method, we stop and wait until all our requests are complete.
> 
> to
> 
> > Using the [hotspot=await file=0]`await` method, you stop and wait until all requests are complete.

I suggest and made the following change

> By using the [hotspot=await file=0]`await` method, the program stops and waits until all requests are complete.

Perhaps saying the program stops is more accurate and less confusing?

For 

> > Delcaring the `values` variable with [hotspot=volatile file=0]`volatile` and instantiating it as a [hotspot=concurrentHashMap file=0]`ConcurrentHashMap` allows it to store system information and safely access it asynchronously from multiple threads.
>
> to
>
> > Declaring the `values` variable with a value of `volatile` and instantiating it as a `ConcurrentHashMap` object allows it to store system information and safely access that information asynchronously from multiple threads.

I have made

> Declaring the `values` variable with the [hotspot=volatile file=0]`volatile` keyword and instantiating it as a [hotspot=concurrentHashMap file=0]`ConcurrentHashMap` object allows it to store system information and safely access that information asynchronously from multiple threads.

I didn't think it was accurate to say the `values` variable had a value of `volatile`, and may have been confusing. 

I have pushed these changes to the lgdev site, please let me know what you think. Thank you.

**Comment by Charlotte-Holt** [comment]
Looks good, thanks @AustinSeto!

**Comment by TiagoF99** [comment]
Verified in QA. Looks good.

**Pull request merged**